packages/Search/src/details/DTK_DetailsTreeTraversal.hpp

```diff
@@ -131,10 +131,12 @@ spatialQuery( BoundingVolumeHierarchy<DeviceType> const &bvh,
 }
 
 // query k nearest neighbours
-template <typename DeviceType, typename Distance, typename Insert>
+template <typename DeviceType, typename Distance, typename Insert,
+          typename Buffer>
 KOKKOS_FUNCTION int
 nearestQuery( BoundingVolumeHierarchy<DeviceType> const &bvh,
-              Distance const &distance, int k, Insert const &insert )
+              Distance const &distance, int k, Insert const &insert,
+              Buffer const &buffer )
```

Comment:
Why is it part of the interface?

Reply:
We need to store up to `k` results during the traversal.  `k` is not known at compile-time and we cannot reasonably come up with an upper bound for it and crash if the number of neighbors queried for exceeds what we anticipated.  Allocating from the kernel is not a very good option either so I chose to pass a buffer with the appropriate size to that function.  That said, it does not have to be a "buffer", we could pass anything that we can push pairs (index, distance) to, and defer all the heap handling to  it.